src/qml/Legend.qml

```diff
@@ -77,214 +84,279 @@ ListView {
       id: line
       anchors.verticalCenter: parent.verticalCenter
       leftPadding: itemPadding
-      spacing: 5
+      spacing: 0
 
       // Collapsed state visual feedback
-      Rectangle {
+      Row {
+        anchors.verticalCenter: parent.verticalCenter
+        height: 24
+        visible: HasChildren
+
+        Item {
           height: 24
-          width: 10
-          color: "transparent"
+          width: HasChildren ? 24 : 0
+          clip: true
           anchors.verticalCenter: parent.verticalCenter
+          visible: HasChildren
 
           Image {
```

Comment:
Can we transform that into a QfToolButton?  That way we don't need the mouse area item, and we can use iconColor to color the icon (instead of relying on two source SVGs).

Reply:
Yes much better, less code and better. thanks.